src/kontrol/foundry.py

```diff
@@ -114,25 +145,63 @@ def cut_point_rules(
             break_on_load_program,
         )
 
-    def _check_rename_pattern(self, cterm: CTerm) -> bool:
-        """Given a CTerm, check if the rule 'FOUNDRY-CHEAT-CODES.rename' is at the top of the K_CELL.
-
-        :param cterm: The CTerm representing the current state of the proof node.
-        :return: `True` if the pattern matches and a custom step can be made; `False` otherwise.
-        """
-        abstract_pattern = KSequence(
+    @property
+    def _rename_pattern(self) -> KSequence:
+        return KSequence(
             [
                 KApply('foundry_rename', [KVariable('###RENAME_TARGET'), KVariable('###NEW_NAME')]),
                 KVariable('###CONTINUATION'),
             ]
         )
-        self._cached_subst = abstract_pattern.match(cterm.cell('K_CELL'))
-        return self._cached_subst is not None
 
-    def _exec_rename_custom_step(self, cterm: CTerm) -> KCFGExtendResult | None:
-        subst = self._cached_subst
-        assert subst is not None
+    @property
+    def _forget_branch_pattern(self) -> KSequence:
+        return KSequence(
+            [
+                KApply('cheatcode_forget', [KVariable('###TERM1'), KVariable('###OPERATOR'), KVariable('###TERM2')]),
+                KVariable('###CONTINUATION'),
+            ]
+        )
 
+    @property
+    def _call_fork_pattern(self) -> KSequence:
+        return KSequence(
+            [
+                KApply(
+                    'callwithcode_check_fork',
+                    [
+                        KVariable('###ACCTFROM'),
+                        KVariable('###ACCTTO'),
+                        KVariable('###ACCTCODE'),
+                        KToken(token='b""', sort=KSort(name='Bytes')),
+                        KVariable('###VALUE'),
+                        KVariable('###APPVALUE'),
+                        KVariable('###ARGS'),
+                        KVariable('###STATIC'),
+                    ],
+                ),
+                KApply('#return___EVM_KItem_Int_Int', [KVariable('###RETURN1'), KVariable('###RETURN2')]),
+                KApply('pc', KVariable('###PC')),
+                KApply('execute', []),
+                KVariable('###CONTINUATION'),
+            ]
+        )
+
+    @property
+    def _sload_fork_pattern(self) -> KSequence:
```

Comment:
Similar to this, I need to handle the rest of access opcodes such as `BALANCE`, `EXTCODECOPY`, `EXTCODEHASH`, `EXTCODESIZE`.